odoo/addons/base/transpiler/transpiler_js.py

```diff
@@ -1,378 +1,393 @@
-        """
-        p = re.compile(r"^(?P<space>\s*)import\s+(?P<path>[^;\n]+)", re.MULTILINE)
-        repl = r"require(\g<path>)"
-        self.content = p.sub(repl, self.content)
-
-    def remove_index(self):
-        """
-        Remove in the paths the /index.js.
-        We want to be able to import a module just trough its directory name if it contains an index.js.
-        So we no longer need to specify the index.js in the paths.
-        """
-        def repl(matchobj):
-            path = matchobj["path"]
-            new_path = path[: path.rfind("/index")] + path[0]
-            return f"require({new_path})"
-        p = re.compile(r"require\s*\(\s*(?P<path>(\".*/index/?\")|('.*/index/?')|(`.*/index/?`))\s*\)", re.MULTILINE)
-        self.content = p.sub(repl, self.content)
-
-    def relative_path_to_module_path(self, path_rel):
-        """
-        Convert the relative path into a module path, which is more generic and fancy.
-
-        :param path_rel: a relative path to the current url. (self.url)
-        :return: module path (@module/...)
-        """
-        url_split = self.url.split("/")
-        path_rel_split = path_rel.split("/")
-        nb_back = len([v for v in path_rel_split if v == ".."]) + 1
-        result = "/".join(url_split[:-nb_back] + [v for v in path_rel_split if not v in ["..", "."]])
-        return self.url_to_module_path(result)
-
-    @staticmethod
-    def is_odoo_module(content):
-        """
-        Detect if the file need to be transpiled.
-        We look for a comment containing @odoo-module.
-
-        :param content: source code
-        :return: is this a odoo module that need transpilation ?
-        """
-        result = re.match(r"/\*\*\s+@odoo-module\s+(alias=(?P<alias>\S+))?\s*\*\*/", content)
-        return bool(result)
-
-    def get_aliased_odoo_define_content(self):
-        """
-        To allow smooth transition between the new system and the legacy one, we have the possibility to
-        defined an alternative module name (an alias) that will act as proxy between legacy require calls and
-        new modules.
-
-        Example:
-        If we have a require call somewhere in the odoo source base being:
-        > vat AbstractAction require("web.AbstractAction")
-        we have a problem when we will have converted to module to ES6: its new name will be more like
-        "web/chrome/abstract_action". So the require would fail !
-        So we add a second small modules, an alias, as such:
-        > odoo.define("web/chrome/abstract_action", function(require) {
-        >  return require('web.AbstractAction')[Symbol.for("default")];
-        > });
-
-        To generate this, change your comment on the top of the file.
+
+def convert_function_and_class_export(content, default=False):
+    """
+    Transpile functions and classes that are being exported.
+
+    before: export function name
+    after: const name = __exports.name = function name
```

Comment:
the docstrings should use proper rST formatting e.g.
```reStructuredText
.. code-block:: javascript

    // before
    export function name
    // after
    const name = __exports.name = function name
```
I also think the example should be *complete*, I was very confused by  this this transformation because the "after" line makes no sense, but this is intended for exporting function or class literals correct? So the transformation is *actually*
```reStructuredText
.. code-block:: javascript

    // before
    export function name() {}
    // after
    const name = __exports.name = function name() {}
```
it's just that the pass doesn't really check for the part after the function name, correct?

Reply:
Yes, when I wrote the comments, I made the decision to only put in the before and after the part that was actively searched by the regex and the modification we applied.

Reply:
@xmo-odoo we're wondering what you mean by "the docstrings should use proper rST format". Why should I respect it since we're in docstring ?

Reply:
Because Python tools (including Sphinx' autodoc) are able to process that markup.

You can see that with e.g. [BaseModel.write](https://github.com/odoo/odoo/blob/0322febbea2cfd2d2fd7aedc42ef03632b17c967/odoo/models.py#L3492): https://www.odoo.com/documentation/14.0/reference/orm.html#odoo.models.Model.write

The box (parameters / raises) is part of the builtin docstring-rendering, but the lists, links, callouts, ... are rendered from rST.

So

    before: export function name

will render as

> before: export function name

but

```reStructuredText
.. code-block:: javascript

    export function name
```

will render as

> ```javascript
> export function name
> ```